corgi/core/models.py

```diff
@@ -710,7 +781,48 @@ def root_components(self) -> models.QuerySet["Component"]:
         return self.filter(ROOT_COMPONENTS_CONDITION)
 
 
-class Component(TimeStampedModel):
+class ComponentTaxonomyMixin(models.Model):
+    """Add component taxonomy fields and methods to save them for Components."""
+
+    # Separate class not really needed, but easier to add in other places
+    # if we add different types of component classes in the future
```

Comment:
Which means we can just do that abstraction in the future as well, no? :slightly_smiling_face:

Reply:
Yep, I pulled it out thinking I was going to reuse it, then realized I was mistaken. So I added the comment instead of putting it back - will fix.